src/Symfony/Component/Yaml/Command/LintCommand.php

```diff
@@ -0,0 +1,194 @@
+        foreach ($files as $file) {
+            $filesInfo[] = $this->validate(file_get_contents($file), $file);
+        }
+
+        return $this->display($input, $output, $io, $filesInfo);
+    }
+
+    private function validate($content, $file = null)
+    {
+        $parser = $this->getParser();
+
+        try {
+            $parser->parse($content);
+        } catch (ParseException $e) {
+            return array('file' => $file, 'valid' => false, 'message' => $e->getMessage());
+        }
+
+        return array('file' => $file, 'valid' => true);
+    }
+
+    private function display(InputInterface $input, OutputInterface $output, SymfonyStyle $io, $files)
+    {
+        switch ($input->getOption('format')) {
+            case 'txt':
+                return $this->displayTxt($output, $io, $files);
+            case 'json':
+                return $this->displayJson($io, $files);
+            default:
+                throw new \InvalidArgumentException(sprintf('The format "%s" is not supported.', $input->getOption('format')));
+        }
+    }
+
+    private function displayTxt(OutputInterface $output, SymfonyStyle $io, $filesInfo)
+    {
+        $errors = 0;
+
+        foreach ($filesInfo as $info) {
+            if ($info['valid'] && $output->isVerbose()) {
+                $io->comment('<info>OK</info>'.($info['file'] ? sprintf(' in %s', $info['file']) : ''));
+            } elseif (!$info['valid']) {
+                ++$errors;
+                $io->text(sprintf('<error> ERROR </error> in %s', $info['file']));
+                $io->text(sprintf('<error> >> %s</error>', $info['message']));
+            }
+        }
+
+        if ($errors === 0) {
+            $io->success(sprintf('All %d YAML files contain valid syntax.', count($filesInfo)));
+        } else {
+            $io->warning(sprintf('%d YAML files have valid syntax and %d contain errors.', count($filesInfo) - $errors, $errors));
+        }
+
+        return min($errors, 1);
+    }
+
+    private function displayJson(OutputInterface $output, $filesInfo)
+    {
+        $errors = 0;
+
+        array_walk($filesInfo, function (&$v) use (&$errors) {
+            $v['file'] = (string) $v['file'];
+            if (!$v['valid']) {
+                ++$errors;
+            }
+        });
+
+        $output->writeln(json_encode($filesInfo, JSON_PRETTY_PRINT));
+
+        return min($errors, 1);
+    }
+
+    protected function isReadable($fileOrDirectory)
+    {
+        return is_readable($fileOrDirectory);
+    }
+
+    protected function getParser()
+    {
+        if (!$this->parser) {
+            $this->parser = new Parser();
+        }
+
+        return $this->parser;
+    }
+
+    protected function findFilesInDirectory($directory)
+    {
+        $iterator = new \RecursiveIteratorIterator(
+            new \RecursiveDirectoryIterator($directory, \FilesystemIterator::SKIP_DOTS | \FilesystemIterator::FOLLOW_SYMLINKS),
+            \RecursiveIteratorIterator::SELF_FIRST
```

Comment:
you should actually use ``LEAVES_ONLY`` instead of ``SELF_FIRST``. You don't need directories in the flattened iterator, only files